### PR TITLE
Upgrade date-fns dep to alpha.25 (unicode tokens)

### PR DESCRIPTION
**Pull request by Rendez**
## Types of changes

Upgrading the date-fns dependency to one of the latest version containing a couple of breaking changes:
- `format` using only Unicode tokens (no more `YYYY` but instead `y`, etc.)
- changes in `locale.localize` to obtain the months.

_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Description

I added tests that cover the code branches that are affected. Since this library almost doesn't have tests, is very hard to be 100% sure that these changes don't introduce side-effects. However, the changes aren't complicated to understand and test. Furthermore, based on the date-fns docs between alpha.7 and alpha.25, the other methods (aside from format and localize.month) haven't changed, so I am confident to not have overlooked anything. I reiterate that the lack of test coverage can't guarantee this 100%. With this in mind, feel free to merge, and afterwards I would recommend covering the rest of the code branches with tests as much as possible.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "react-date-range",
-  "version": "1.0.0-beta",
+  "name": "@researchgate/react-date-range",
```
Comment (keremciu) [inline package.json]: what's the case here?

Comment (Rendez) [reply package.json]: Oops, that's our versioned release. I can correct that.

Comment (Rendez) [reply package.json]: @keremciu Done. I set the name back.

**Comment by gsouf** [comment]
Up! still broken with date-fns latest :)

**Comment by gsouf** [comment]
@onurkerimov sorry to bug you but can we have an update on this one?

**Comment by keremciu** [comment]
@gsouf are you using version 1.0.3?

**Comment by gsouf** [comment]
@keremciu yes I do, and I confirm that `react-date-range 1.0.3` is not compatible with `date-fns 2.14.0` due to their breaking change in the way they manage format tokens

**Comment by keremciu** [comment]
> @keremciu yes I do, and I confirm that `react-date-range 1.0.3` is not compatible with `date-fns 2.14.0` due to their breaking change in the way they manage format tokens in version 2

yep yep that's why peer dependencies are referring to 2.0.0, it would be great to have a PR for it but this one is not the one :/

**Comment by gsouf** [comment]
@keremciu would you be fine if I provided a PR to be compatible with date-fns latest? 

Note that would be a bc break so you will proabably want to bump react-date-range to version 2.0

**Comment by keremciu** [comment]
@gsouf would be nice! please go for it :) we can bump to 1.1.0 while it's not that much big change for users, just updating fns

**Comment by gsouf** [comment]
@keremciu I probably made something wrong with my package yesterday because testing today works well.

I think you can close this PR now

**Comment by keremciu** [comment]
> @keremciu I probably made something wrong with my package yesterday because testing today works well.
> 
> I think you can close this PR now
> 
> Thanks for your feedback on this one

great to hear! thanks for the heads up :)

I think this one was unrelevant while the new API with 1.0.3 is covering changes

**Comment by gsouf** [comment]
@keremciu can I still advise to change the peer-dependencie on date fns to be the same as dev deps? ie `^2.8.1`. Would avoid some possible confusions

**Comment by keremciu** [comment]
yep, we're discussing about ownership of this repository, there should be updates regarding API changes then we can be sure about which versions are totally compatible

**Pull request closed without merging**